### PR TITLE
Add seperate method for import_shipping_carriers

### PR DESCRIPTION
**Pull request by cerebro1**
https://3.basecamp.com/3194017/buckets/421793/todos/121516046

### CODE REVIEW

**Comment by cerebro1** [comment]
not_implement_error

File: channel.py [hunk 1]
```diff
@@ -251,15 +251,22 @@ def get_current_channel(cls):
...
         Create shipping carriers by importing data from external channel.
 
-        Since external channels are implemented by downstream modules, it is
-        the responsibility of those channels to reuse this method or call super.
-
         :param instances: Active record list of magento instances
         """
+        pass
+
+    @classmethod
```
Comment (riteshshrv) [inline channel.py]: remove this

File: channel.py [hunk 2]
```diff
@@ -251,15 +251,22 @@ def get_current_channel(cls):
...
+    def import_shipping_carriers(self):
+        '''
+        Create shipping carriers by importing data from external channel.
+
+        Since external channels are implemented by downstream modules, it is
+        the responsibility of those channels to reuse this method or call super.
+        '''
         raise NotImplementedError(
             "This feature has not been implemented."
         )
```
Comment (riteshshrv) [inline channel.py]: include channel name in this error

File: channel.py [hunk 3]
```diff
@@ -251,15 +251,22 @@ def get_current_channel(cls):
 
     @classmethod
     @ModelView.button
-    def import_shipping_carriers(cls, channels):
+    def import_shipping_carriers_button(cls, channels):
         """
         Create shipping carriers by importing data from external channel.
```
Comment (riteshshrv) [inline channel.py]: `Import shipping carriers by calling instance method 'import_shipping_carriers'  implemented by downstream modules`

File: channel.py [hunk 4]
```diff
@@ -251,15 +251,22 @@ def get_current_channel(cls):
...
 
-        Since external channels are implemented by downstream modules, it is
-        the responsibility of those channels to reuse this method or call super.
-
         :param instances: Active record list of magento instances
         """
+        pass
```
Comment (riteshshrv) [inline channel.py]:
```python
for channel in channels:
    channel.import_shipping_carriers()
```

Comment (riteshshrv) [inline channel.py]: you can't just `pass` this method; who is going to call the instance method?

File: channel.py [hunk 5]
```diff
@@ -251,17 +251,25 @@ def get_current_channel(cls):
...
     @ModelView.button
-    def import_shipping_carriers(cls, channels):
+    def import_shipping_carriers_button(cls, channels):
         """
         Create shipping carriers by importing data from external channel.
 
-        Since external channels are implemented by downstream modules, it is
-        the responsibility of those channels to reuse this method or call super.
-
         :param instances: Active record list of magento instances
```
Comment (tarunbhardwaj) [inline channel.py]: channels: Active record list of sale channels

**Comment by riteshshrv** [comment]
LGTM

**Comment by tarunbhardwaj** [comment]
change in line 189. and now method is instance method you have to update in trytond-magento also

**Pull request closed without merging**